### PR TITLE
[EBPF] Add kmt.selftest task

### PR DESCRIPTION
**Pull request by gjulianm**
### What does this PR do?

Adds a new `kmt.selftest` task that can be used to ensure correctness of the KMT code and installation. 

For now only basic tests are added, more tests can follow in subsequent PRs.

### Motivation

Facilitating development and ensuring consistency when refactoring KMT code (e.g., future cross-compiling support).

### Additional Notes

Some arguments have been added to other tasks to facilitate automated testing:

- `--yes` argument to  `kmt.gen-config` so that changes to the config are applied without confirmation
- `--compile-only` argument to `kmt.prepare` so that the consistency of the generated binaries can be checked without having to transfer them to VMs.

### CODE REVIEW

File: tasks/kernel_matrix_testing/selftest.py [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+
+from tasks.kernel_matrix_testing.platforms import get_platforms
+from tasks.kernel_matrix_testing.tool import error, full_arch, get_binary_target_arch, info, warn
+from tasks.kernel_matrix_testing.types import Component
+from tasks.kernel_matrix_testing.vars import arch_ls
+
+SelftestResult = Tuple[Optional[bool], str]
+
+
+def selftest_pulumi(ctx: Context, _: bool) -> SelftestResult:
```
Comment (val06) [inline tasks/kernel_matrix_testing/selftest.py:18]: consider fixing the required version / versions range in the selftest (see inv [setup](https://github.com/DataDog/datadog-agent/blob/eab9ad9595ca2caccc45d544b1243b6c4f949479/tasks/setup.py#L27) for example)

**Review by val06 — CHANGES_REQUESTED** [review]
reviewed

File: tasks/kernel_matrix_testing/selftest.py [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+            if "image_version" not in data:
+                return False, f"Image {image} does not have image_version field"
+            image_vers.add(data["image_version"])
+
+    if len(image_vers) != 1:
+        return False, f"Multiple image versions found: {image_vers}"
+
+    res = ctx.run("inv -e kmt.ls", hide=True, warn=True)
+    if res is None or not res.ok:
+        return False, "Cannot run inv -e kmt.ls, platforms.json file might be incorrect"
```
[the remark this answers is not in the document]
Comment (gjulianm) [reply tasks/kernel_matrix_testing/selftest.py:48]: In this case I want this command to be run as part of the self-test suite, basically to ensure that the code that builds all the images and distribution mappings works correctly. The output here is not as important.

File: tasks/kernel_matrix_testing/selftest.py (hunk 1 above)
Comment (gjulianm) [reply tasks/kernel_matrix_testing/selftest.py:18]: Do we have a required version range for pulumi? I am looking at the [test-infra-definitions setup](https://github.com/DataDog/test-infra-definitions/blob/main/tasks/setup.py#L72) and it seems we just download the most recent version.

File: tasks/kmt.py [hunk 3]
```diff
@@ -594,6 +600,7 @@ def prepare(
     packages=None,
     verbose=True,
     ci=False,
+    compile_only=True,
```
[the remark this answers is not in the document]
Comment (gjulianm) [reply tasks/kmt.py:603]: Typo, fixed now.

File: tasks/kmt.py [hunk 4]
```diff
@@ -549,6 +554,7 @@ def kmt_secagent_prepare(
     packages: str | None = None,
     verbose: bool = True,
     ci: bool = True,
+    compile_only: bool = True,
```
[the remark this answers is not in the document]
Comment (gjulianm) [reply tasks/kmt.py:557]: Typo, fixed now.

**Review by val06 — APPROVED** [review]

**Comment by gjulianm** [comment]
/trigger-ci --variable RUN_ALL_BUILDS=true --variable RUN_KITCHEN_TESTS=true --variable RUN_E2E_TESTS=on --variable RUN_UNIT_TESTS=on --variable RUN_KMT_TESTS=on

**Comment by gjulianm** [comment]
/trigger-ci --variable RUN_ALL_BUILDS=true --variable RUN_KITCHEN_TESTS=true --variable RUN_E2E_TESTS=on --variable RUN_UNIT_TESTS=on --variable RUN_KMT_TESTS=on

**Pull request merged**